docs/source/algo_walkthrough.rst

```diff
@@ -0,0 +1,360 @@
+The ``FastCorrelativeScanMatcher`` relies on a "*Branch and bound*" mechanism to work at different grid resolutions and efficiently eliminate incorrect matchings.
+This mechanism is extensively presented in the Cartographer paper presented earlier in this document.
+It works on an exploration tree whose depth can be controled.
+
+.. code-block:: lua
+
+    POSE_GRAPH.constraint_builder.fast_correlative_scan_matcher.branch_and_bound_depth
+    POSE_GRAPH.constraint_builder.fast_correlative_scan_matcher_3d.branch_and_bound_depth
+    POSE_GRAPH.constraint_builder.fast_correlative_scan_matcher_3d.full_resolution_depth
+
+Once the ``FastCorrelativeScanMatcher`` has a good enough proposal (above a minimum score of matching), it is then fed into a Ceres Scan Matcher to refine the pose.
+
+.. code-block:: lua
+
+    POSE_GRAPH.constraint_builder.min_score
+    POSE_GRAPH.constraint_builder.ceres_scan_matcher_3d
+    POSE_GRAPH.constraint_builder.ceres_scan_matcher
+
+When Cartographer runs *the optimization problem*, Ceres is used to rearrange submaps according to multiple "*residuals*".
+A residual is essentially a cost function with an associated weight.
+The global optimization has cost functions to take into account plenty of data sources: the global (loop closure) constraints, the non-global (matcher) constraints, the IMU acceleration and rotation measurements, the local SLAM rough pose estimations, an odometry source or a fixed frame (such as a GPS system).
+The weights and Ceres options can be configured as described in the Local SLAM section.
+
+.. code-block:: lua
+
+    POSE_GRAPH.constraint_builder.loop_closure_translation_weight
+    POSE_GRAPH.constraint_builder.loop_closure_rotation_weight
+    POSE_GRAPH.matcher_translation_weight
+    POSE_GRAPH.matcher_rotation_weight
+    POSE_GRAPH.optimization_problem.*_weight
+    POSE_GRAPH.optimization_problem.ceres_solver_options
+
+.. note::
+
+    One can find useful information about the residuals used in the optimization problem by toggling ``POSE_GRAPH.max_num_final_iterations``
+
+As part of its IMU residual, the optimization problem gives some flexibility to the IMU pose and, by default, Ceres is free to optimize the extrinsic calibration between your IMU and tracking frame.
+If you don't trust your IMU pose, the results of Ceres' global optimization can be logged and used to improve your extrinsic calibration.
+If Ceres doesn't optimize your IMU pose correctly and you trust your extrinsic calibration enough, you can make this pose constant.
+
+.. code-block:: lua
+
+    POSE_GRAPH.optimization_problem.log_solver_summary
+    POSE_GRAPH.optimization_problem.use_online_imu_extrinsics_in_3d
+
+In residuals, the influence of outliers is handled by a **Huber loss** function configured with a certain a Huber scale.
+The bigger the Huber scale, the more outliers will be taken into account.
```

Comment:
`The bigger the Huber scale, the higher is the impact of (potential) outliers.`

Reply:
You can also link his comment from Ceres ;)
https://github.com/ceres-solver/ceres-solver/blob/0d3a84fce553c9f7aab331f0895fa7b1856ef5ee/include/ceres/loss_function.h#L172